docscan/session/create/check/liveness.go

```diff
@@ -1,29 +1,44 @@
 package check
 
 import (
+	"encoding/json"
+
 	"github.com/getyoti/yoti-go-sdk/v3/docscan/constants"
 )
 
 // RequestedLivenessCheck requests creation of a Liveness Check
 type RequestedLivenessCheck struct {
-	RequestedCheck                          //TODO: pointer in embedded type?
-	Config         *RequestedLivenessConfig `json:"config"`
+	checkType string
+	config    RequestedLivenessConfig
 }
 
-// NewRequestedLivenessCheck creates a new Document Authenticity Check
-func NewRequestedLivenessCheck(config *RequestedLivenessConfig) *RequestedLivenessCheck {
-	return &RequestedLivenessCheck{
-		RequestedCheck: RequestedCheck{
-			Type: constants.Liveness,
-		},
-		Config: config,
-	}
+// Type is the type of the Requested Check
+func (c RequestedLivenessCheck) Type() string {
+	return c.checkType
+}
+
+// Config is the configuration of the Requested Check
+func (c RequestedLivenessCheck) Config() RequestedCheckConfig {
+	return RequestedCheckConfig(
+		c.config,
+	)
+}
+
+// MarshalJSON returns the JSON encoding
+func (c RequestedLivenessCheck) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Type   string               `json:"type"`
+		Config RequestedCheckConfig `json:"config,omitempty"`
+	}{
+		Type:   c.Type(),
+		Config: c.Config(),
+	})
 }
 
 // RequestedLivenessConfig is the configuration applied when creating a Liveness Check
 type RequestedLivenessConfig struct {
-	RequestedCheckConfig
-	MaxRetries   *int   `json:"max_retries"`
+	// RequestedCheckConfig
```

Comment:
Did you mean to remove this?

Reply:
Yes I meant to remove, have done that in d70a6d2